### PR TITLE
Redux user state

### LINKED ISSUE #51: Redux userState

### PR DESCRIPTION
**Pull request by eckernas**
closes #51

### CODE REVIEW

File: src/model/redux/userState.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        });
+        const updateUser = await UserModel.updateUser(user);
+        dispatch({
+            type: UserActionTypes.UPDATE_USER_DATA,
+            payload: updateUser
+        });
+        dispatch({
+            type: UserActionTypes.UPDATE_USER_LOADING,
+            payload: false
+        });
```
Comment (AdamJonsson) [inline src/model/redux/userState.tsx:87]: Wrap this with a batch: https://react-redux.js.org/api/batch

File: src/model/redux/userState.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        const newUser = await UserModel.addUser(user)
+        dispatch({
+            type: UserActionTypes.ADD_USER_DATA,
+            payload: newUser
+        });
+        dispatch({
+            type: UserActionTypes.UPDATE_USER_LOADING,
+            payload: false
+        });
```
Comment (AdamJonsson) [inline src/model/redux/userState.tsx:59]: Wrap this with a batch: https://react-redux.js.org/api/batch

File: src/model/redux/userState.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    loading: boolean
+}
+
+/*Available actions to for example change the state*/
+export enum UserActionTypes {
+    ADD_USER_DATA = 'ADD_USER_DATA',
+    REMOVE_USER_DATA = 'REMOVE_USER_DATA',
+    UPDATE_USER_DATA = 'UPDATE_USER_DATA',
+    UPDATE_USER_LOADING = 'UPDATE_USER_LOADING'
+}
```
Comment (AdamJonsson) [inline src/model/redux/userState.tsx:18]: We need another login action. I like really like the way you reuse the different actions. I think that you can do the same with the new login action

**Comment by eckernas** [comment]
I have updated the state now! I am still a bit unsure if this was what you meant. @AdamJonsson

**Pull request merged**